src/main/java/org/opensearch/security/ssl/SecureSSLSettings.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.opensearch.security.ssl;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.opensearch.common.settings.SecureSetting;
+import org.opensearch.common.settings.SecureString;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+import static org.opensearch.security.ssl.util.SSLConfigConstants.DEFAULT_STORE_PASSWORD;
+
+/**
+ * Container for secured settings (passwords for certs, keystores) and the now deprecated original settings
+ */
+public final class SecureSSLSettings {
+    private static final String SECURE_SUFFIX = "_secure";
+    private static final String PREFIX = "plugins.security.ssl";
+    private static final String HTTP_PREFIX = PREFIX + ".http";
+    private static final String TRANSPORT_PREFIX = PREFIX + ".transport";
+
+    public enum SSLSetting {
```

Comment:
Nicely done!

Metrics: this enum removed +50 lines additional of code to be deleted, and 100 less lines of code to be added.  Yay for easier to maintain codebases.